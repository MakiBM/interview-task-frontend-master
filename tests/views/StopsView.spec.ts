import { describe, it, expect, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import { createStore } from 'vuex'
import Component from '../../src/views/StopsView.vue'

describe('Stops View Snapshot Tests', () => {
  const createMockStore = () => {
    return createStore({
      getters: {
        getStopNamesBySearchQuery: () => ['Stop 1', 'Stop 2', 'Stop 3', 'Line A'], // Mock list of stops
      }
    })
  }

  it('matches the snapshot with an empty search query', async () => {
    const store = createMockStore()
    const wrapper = mount(Component, {
      global: {
        plugins: [store],
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
