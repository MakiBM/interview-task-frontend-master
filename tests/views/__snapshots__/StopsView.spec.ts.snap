// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Stops View Snapshot Tests > matches the snapshot with an empty search query 1`] = `
"<div class="stops" data-v-aed941f8="">
  <div class="card border-0 h-100" data-v-683929d4="" data-v-aed941f8="">
    <div class="card-header" data-v-683929d4="">
      <div class="p-3 px-2" data-v-683929d4="">
        <!-- Card header -->
        <div class="search mb-4" data-v-aed941f8="">
          <div class="form-input" for="search" title="search" data-v-28dbba01="" data-v-aed941f8=""><input class="form-input__input" type="text" placeholder="Search..." id="search" data-v-28dbba01=""><!-- Use for icon --><svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg" data-v-aed941f8="">
              <path fill-rule="evenodd" clip-rule="evenodd" d="M1.83325 7.33337C1.83325 4.29581 4.29569 1.83337 7.33325 1.83337C10.3708 1.83337 12.8333 4.29581 12.8333 7.33337C12.8333 10.3709 10.3708 12.8334 7.33325 12.8334C4.29569 12.8334 1.83325 10.3709 1.83325 7.33337ZM7.33325 0.833374C3.7434 0.833374 0.833252 3.74352 0.833252 7.33337C0.833252 10.9232 3.7434 13.8334 7.33325 13.8334C8.9482 13.8334 10.4256 13.2444 11.5624 12.2696L13.6464 14.3536C13.8416 14.5489 14.1582 14.5489 14.3535 14.3536C14.5487 14.1583 14.5487 13.8418 14.3535 13.6465L12.2695 11.5625C13.2443 10.4257 13.8333 8.94832 13.8333 7.33337C13.8333 3.74352 10.9231 0.833374 7.33325 0.833374Z" fill="#9A9DA4" data-v-aed941f8=""></path>
            </svg></div>
        </div>
        <div class="subtitle semibold" data-v-aed941f8=""><span data-v-aed941f8="">Bus Stops</span>&nbsp; <span role="button" data-v-aed941f8=""><svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg" style="transform: rotate(180deg);" data-v-aed941f8=""><path d="M10.6668 8.66665L8.00004 11.3333M8.00004 11.3333L5.33337 8.66665M8.00004 11.3333L8.00004 4.66665M6.6667 14.6666H9.33337C11.2002 14.6666 12.1336 14.6666 12.8467 14.3033C13.4739 13.9838 13.9838 13.4738 14.3034 12.8466C14.6667 12.1336 14.6667 11.2002 14.6667 9.33332V6.66665C14.6667 4.7998 14.6667 3.86638 14.3034 3.15334C13.9838 2.52614 13.4739 2.0162 12.8467 1.69662C12.1336 1.33331 11.2002 1.33331 9.33338 1.33331H6.66671C4.79987 1.33331 3.86644 1.33331 3.15341 1.69662C2.5262 2.0162 2.01626 2.52614 1.69669 3.15334C1.33337 3.86638 1.33337 4.7998 1.33337 6.66664V9.33331C1.33337 11.2002 1.33337 12.1336 1.69669 12.8466C2.01626 13.4738 2.5262 13.9838 3.15341 14.3033C3.86644 14.6666 4.79986 14.6666 6.6667 14.6666Z" stroke="#63666E" stroke-linecap="round" stroke-linejoin="round"></path></svg></span></div>
      </div>
    </div>
    <div class="card-body" data-v-683929d4="">
      <ul class="list-group list-group-flush" data-v-683929d4="">
        <li class="list-group-item list-group-item-action py-3" role="" data-v-683929d4="">
          <p class="px-2 mb-0" data-v-683929d4=""><small data-v-683929d4="">Stop 1</small></p>
        </li>
        <li class="list-group-item list-group-item-action py-3" role="" data-v-683929d4="">
          <p class="px-2 mb-0" data-v-683929d4=""><small data-v-683929d4="">Stop 2</small></p>
        </li>
        <li class="list-group-item list-group-item-action py-3" role="" data-v-683929d4="">
          <p class="px-2 mb-0" data-v-683929d4=""><small data-v-683929d4="">Stop 3</small></p>
        </li>
        <li class="list-group-item list-group-item-action py-3" role="" data-v-683929d4="">
          <p class="px-2 mb-0" data-v-683929d4=""><small data-v-683929d4="">Line A</small></p>
        </li>
      </ul>
    </div>
  </div>
</div>"
`;
